code/common/clock.cc

```diff
@@ -55,37 +75,63 @@ namespace common::time
   {
     reset();
 
-    m_start_cycle = current_cycle_count();
+    #if defined(_WIN32)
+      m_start_cycle = current_cycle_count();
+    #elif defined(__linux__)
+      this->m_start_time = current_timestamp();
+    #else
+      static_assert(false, "Not implemented");
+    #endif
     m_started = true;
   }
 
   void stopwatch::stop()
   {
     if (m_started)
+    #if defined(_WIN32)
       m_end_cycle = current_cycle_count();
-
+    #elif defined(__linux__)
+      this->m_end_time = current_timestamp();
+    #else
+      static_assert(false, "Not implemented");
+    #endif
     m_started = false;
   }
 
   void stopwatch::reset()
   {
     m_started = false;
-    m_start_cycle = 0;
-    m_end_cycle = 0;
+    #if defined(_WIN32)
+        m_start_cycle = 0;
+        m_end_cycle = 0;
+    #elif defined(__linux__)
+        m_start_time = 0.;
+        m_end_time   = 0.;
+    #else
+        static_assert(false, "Not implemented");
+    #endif
   }
 
+  #if defined(_WIN32)
   cycles_type stopwatch::delta_cycles()
   {
     if (m_started)
       return current_cycle_count() - m_start_cycle;
 
     return m_end_cycle - m_start_cycle;
   }
+  #endif
 
   seconds_type stopwatch::delta_seconds()
   {
-    const auto cycles = delta_cycles();
-
-    return cycles_to_seconds(cycles);
+    #if defined(_WIN32)
+        const auto cycles = delta_cycles();
+
+        return cycles_to_seconds(cycles);
+    #elif defined(__linux__)
+        return this->m_end_time - this->m_start_time;
```

Comment:
No `this->` prefix please

Reply:
Addressed in 95a120ab595e8373fb27e8519944e828bcb9d8fd